File: roles/es_demo/files/opt/splunk/etc/apps/SplunkEnterpriseSecurityInstaller/default/src/etc/apps/Splunk_TA_nix/appserver/modules/TA_Unix_FTR/TA_Unix_FTR.py
# Copyright 2011 Splunk, Inc.                                                                       
#                                                                                                        
#   Licensed under the Apache License, Version 2.0 (the "License");                                      
#   you may not use this file except in compliance with the License.                                     
#   You may obtain a copy of the License at                                                              
#                                                                                                        
#       http://www.apache.org/licenses/LICENSE-2.0                                                       
#                                                                                                        
#   Unless required by applicable law or agreed to in writing, software                                  
#   distributed under the License is distributed on an "AS IS" BASIS,                                    
#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.                             
#   See the License for the specific language governing permissions and                                  
#   limitations under the License.    

import json
import logging
import os
import shutil

import cherrypy
import splunk
from splunk.appserver.mrsparkle.lib import jsonresponse
from splunk.appserver.mrsparkle.lib import util as app_util
import controllers.module as module

from splunk.models.app import App

logger = logging.getLogger('splunk.module.TA_Unix_FTR')

CONFLICT_APPS = ['unix', 'unix_new']
STATIC_APP = 'Splunk_TA_nix'

class TA_Unix_FTR(module.ModuleHandler):

    def generateResults(self, **kwargs):

        app_name = kwargs.get('client_app', STATIC_APP)
        legacy_setup = os.path.join(app_util.get_apps_dir(), app_name,
                                    'default', 'setup.xml')
                                    
        if os.path.exists(legacy_setup):
            shutil.move(legacy_setup, legacy_setup + '.bak')
            logger.info('disabled legacy setup.xml for %s' % app_name)
            
        for app in App.all():
            if app.name in CONFLICT_APPS and not app.is_disabled:
                return self.render_json({'is_conflict': True, 'app_label': app.label})

        return self.render_json({})
      
    def render_json(self, response_data, set_mime='text/json'):
        ''' 
        clone of BaseController.render_json, which is
        not available to module controllers (SPL-43204)
        '''

        cherrypy.response.headers['Content-Type'] = set_mime

        if isinstance(response_data, jsonresponse.JsonResponse):
            response = response_data.toJson().replace("</", "<\\/")
        else:
            response = json.dumps(response_data).replace("</", "<\\/")

        # Pad with 256 bytes of whitespace for IE security issue. See SPL-34355
        return ' ' * 256  + '\n' + response

File: roles/es_demo/files/opt/splunk/etc/apps/SplunkEnterpriseSecurityInstaller/default/src/etc/apps/Splunk_TA_nix/appserver/modules/TA_Unix_FTR/TA_Unix_FTR.js
// Copyright 2011 Splunk, Inc.                                      
//
//   Licensed under the Apache License, Version 2.0 (the "License");         
//   you may not use this file except in compliance with the License.        
//   You may obtain a copy of the License at                                 
//                                                                           
//       http://www.apache.org/licenses/LICENSE-2.0                          
//                                                                           
//   Unless required by applicable law or agreed to in writing, software     
//   distributed under the License is distributed on an "AS IS" BASIS,       
//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//   See the License for the specific language governing permissions and     
//   limitations under the License.    

Splunk.namespace("Module");
Splunk.Module.TA_Unix_FTR= $.klass(Splunk.Module, {

    COLLISION_WARN: '<p class="popupText">The app "%s" is installed on this system.</p><p class="popupText">Splunk for Unix Technical Add-on and the "%s" app cannot exist together on the same Splunk instance.</p><p class="popupText">Please click on Manage Apps to disable the conflicting app, then remove "%s" from $SPLUNK_HOME/etc/apps and restart Splunk.</p>',
    
    initialize: function($super, container) {
        $super(container);
        this.logger = Splunk.Logger.getLogger("ta_unix_ftr.js");
        this.messenger = Splunk.Messenger.System.getInstance();
        this.popupDiv = $('.ftrPopup', this.container).get(0);
        this.getResults();
    },

    renderResults: function(response, turbo) {
        if (response.is_conflict && response.is_conflict===true) {
            this.popupDiv.innerHTML = this.COLLISION_WARN.replace(/%s/g, response.app_label);
            this.popup = new Splunk.Popup(this.popupDiv, {
                cloneFlag: false,
                title: _("Unsupported Configuration"),
                pclass: 'configPopup',
                buttons: [
                    {
                        label: _("Manage Apps"),
                        type: "primary",
                        callback: function(){
                            Splunk.util.redirect_to('manager/' + Splunk.util.getCurrentApp() + '/apps/local');                                                    
                        }.bind(this)
                    }       
                ]       
           });      
        } 
    }

});

